src/Plugin/Block/ListFacetPages.php

```diff
@@ -21,38 +22,47 @@ class ListFacetPages extends AbstractConfiguredBlockBase {
    */
   public function build() {
     $block = [];
-    $fields = $this->configFactory->get('islandora_solr_facet_pages.settings')->get('islandora_solr_facet_pages_fields_data');
-
-    $items = [];
-    foreach ($fields as $value) {
-      $items[] = [
-        '#markup' => Link::createFromRoute(
-          $value['label'],
-          'islandora_solr_facet_pages.callback',
-          ['path' => $value['path']],
-          ['attributes' => ['title' => $value['label']]]
-        )->toString(),
-      ];
-    }
 
-    if (!empty($items)) {
+    $config = $this->configFactory->get('islandora_solr_facet_pages.settings');
+    $fields = $config->get('islandora_solr_facet_pages_fields_data');
+
+    $cache_meta = CacheableMetadata::createFromRenderArray($block)
```

Comment:
There not being any context to preserve here perhaps we should not create from render array.